### PR TITLE
Drop OpenSSL submodule in favor of Mbed TLS

### PR DESCRIPTION
**Pull request by splasky**
With below reason so we remove openssl module inside TA-endpoint:
1. Mbedtls is more tiny and more suitable for IOT devices.
2. http_conn has already changed to mbedtls.
3. Less dependency can let us more easily migrate into legato framework.

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -28,7 +28,18 @@
 #define ADDR_LOG_PATH "addr_log.log"
 #endif
 
+const uint8_t iv_global[16] = {164, 3,  98, 193, 52,  162, 107, 252,
```
Comment (jserv) [inline main.c]: You should make comments around this lines.

**Comment by jserv** [comment]
The subject of pull request was changed for precise wording. Please revise git commit messages as well.

File: main.c (hunk 1 above)
Comment (howjmay) [reply main.c]: This is an IV for testing purpose. It may be better to give corresponding comment here

**Review by howjmay — CHANGES_REQUESTED** [review]

File: main.c [hunk 2]
```diff
@@ -114,26 +122,59 @@ int main(int argc, char *argv[]) {
...
+#else
+      snprintf(raw_msg, raw_msg_len, MSG, next_addr);
+#endif
       printf("Raw Message: %s\n", raw_msg);
-      encrypt(raw_msg, strlen(raw_msg), ciphertext, &ciphertext_len, iv);
-
+      uint8_t private_key[AES_BLOCK_SIZE * 2] = { 0 };
+      char id[IMSI_LEN + 1] = { 0 };
+#ifndef DEBUG
+      if(get_aes_key(private_key) != 0){
```
Comment (jserv) [inline main.c]: Follow consistent coding style.

File: main.c [hunk 3]
```diff
@@ -114,26 +122,59 @@ int main(int argc, char *argv[]) {
...
+                             13,  159, 135, 113, 49,  209, 58,  68};
+
+    memcpy(id, device_id, 16);
+    memcpy(private_key, key, 16);
+    memcpy(iv, iv_global, 16);
+#endif
+      ciphertext_len = encrypt(raw_msg, strlen(raw_msg),
+                           ciphertext, 1024,
+                           iv, private_key, id);
+      if(ciphertext_len == 0){
```
Comment (jserv) [inline main.c]: Ditto.

**Review by jserv — CHANGES_REQUESTED** [review]
You have to explain the motivation and considerations to migrate to Mbed TLS.

**Comment by splasky** [comment]
Done. Please review again.

**Comment by jserv** [comment]
In GitHub user interface, you can press the button in "Reviewers" for asking further review.

File: main.c [hunk 4]
```diff
@@ -68,15 +70,21 @@ void send_https_msg(char const *const host, char const *const port,
   free(parser);
 }
 
-int main(int argc, char *argv[]) {
-  int ret, size;
+void log_address(char *next_addr) {
+  FILE *fp;
+  char addr_log[ADDR_LEN + 3];
+  // Append the next address to the address log file
+  fp = fopen(ADDR_LOG_PATH, "a");
```
Comment (jserv) [inline main.c:77]: Why is file "a" touched?

Comment (splasky) [reply main.c:77]: "a" is attach mode to log address into log file from line 89.

Comment (jserv) [reply main.c:77]: ``a''  is Open for writing, that might be considered as harmful when you can not ensure the output directory. That is, it is difficult to understand the problem resulting from permissions and/or damage.

Comment (jserv) [reply main.c:77]: Thus, we expect fine-grained checks on file creation.

**Review by jserv — APPROVED** [review]
+1

**Review by howjmay — APPROVED** [review]

**Pull request merged**